zrml/styx/src/lib.rs

```diff
@@ -0,0 +1,116 @@
+        pallet_prelude::*,
+        traits::{Currency, NamedReservableCurrency},
+    };
+    use frame_system::pallet_prelude::*;
+    use sp_runtime::SaturatedConversion;
+    use zeitgeist_primitives::types::Balance;
+
+    use crate::weights::WeightInfoZeitgeist;
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        /// The origin that is allowed to destroy markets.
+        type SetBurnAmountOrigin: EnsureOrigin<Self::Origin>;
+
+        /// Pallet event type
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+        /// Balances pallet
+        type Currency: NamedReservableCurrency<Self::AccountId, ReserveIdentifier = [u8; 8]>;
+
+        /// Weights generated by benchmarks
+        type WeightInfo: WeightInfoZeitgeist;
+    }
+
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    pub struct Pallet<T>(_);
+
+    /// Keep track of crossings. Accounts are only able to cross once.
+    #[pallet::storage]
+    pub type Crossings<T: Config> = StorageMap<_, Blake2_128Concat, T::AccountId, bool>;
+
+    #[pallet::type_value]
+    pub fn DefaultBurnAmount<T: Config>() -> Balance {
+        (zeitgeist_primitives::constants::BASE * 100).saturated_into()
+    }
+
+    /// An extra layer of pseudo randomness.
+    #[pallet::storage]
+    pub type BurnAmount<T: Config> = StorageValue<_, Balance, ValueQuery, DefaultBurnAmount<T>>;
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        /// A account crossed and claimed their right to create their avatar.
+        AccountCrossed(T::AccountId, Balance),
+        /// The crossing fee was changed.
+        CrossingFeeChanged(Balance),
+    }
+
+    #[pallet::error]
+    pub enum Error<T> {
+        /// Account does not have enough balance to cross.
+        FundDoesNotHaveEnoughFreeBalance,
+        /// Account has already crossed.
+        HasAlreadyCrossed,
+    }
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        /// Burns ZTG(styx.burnAmount()) to cross, granting the ability to claim your zeitgeist avatar.
+        #[pallet::weight(T::WeightInfo::cross())]
+        pub fn cross(origin: OriginFor<T>) -> DispatchResult {
+            let who = ensure_signed(origin)?;
+
+            if Crossings::<T>::contains_key(&who) {
+                Err(Error::<T>::HasAlreadyCrossed)?;
+            }
+
+            let amount = BurnAmount::<T>::get().saturated_into();
+            let free = T::Currency::free_balance(&who);
+
+            if free < amount {
+                Err(Error::<T>::FundDoesNotHaveEnoughFreeBalance)?;
+            }
+
+            T::Currency::slash(&who, amount);
+            Crossings::<T>::insert(&who, true);
+
+            Self::deposit_event(Event::AccountCrossed(who, amount.saturated_into()));
+
+            Ok(())
+        }
+
+        /// Set the burn amount. Needs 50% council vote.
```

Comment:
"Needs 50% council vote" is not correct. That's how we implement it in the runtime. But here, the calling origin must be `SetBurnAmountOrigin`, that's it.

Reply:
I wrote that because the message shows in the polkadotjs ui to the user giving some hint that its not callable by anon. Should it read differently you think?

Reply:
Yes, the pallet documentation should reflect what's actually happening in the pallet, not what's happening in the runtime that uses the pallet.